### ISSUE TITLE
[Bug]: [range_search] Range search failed reporting " [UnexpectedError] Assert "(index < size_)" when input valid "radius" and "range_filter" parameters

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#22348]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: milvus-latest(2.3 latest)
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka):    rocksmq
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.3.0.dev38
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Range search failed reporting " [UnexpectedError] Assert "(index < size_)" when input valid "radius" and "range_filter" parameters
```
>>> default_search_params = {"metric_type": "L2", "params": {"nprobe": 10, "radius": 10, "range_filter": 5}}
>>> res1 = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0")
RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=1, reason=stream operation failed: [UnexpectedError] Assert "(index < size_)" at /go/src/github.com/milvus-io/milvus/internal/core/src/segcore/ConcurrentVector.h:51
)>, <Time:{'RPC start': '2023-02-22 09:37:14.306915', 'RPC error': '2023-02-22 09:37:14.319309'}>
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/binbin/towhee-test-env-3.8-0.8.1/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 614, in search
    res = conn.search(self._name, data, anns_field, param, limit, expr,
  File "/binbin/towhee-test-env-3.8-0.8.1/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler
    raise e
  File "/binbin/towhee-test-env-3.8-0.8.1/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler
    return func(*args, **kwargs)
  File "/binbin/towhee-test-env-3.8-0.8.1/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler
    ret = func(self, *args, **kwargs)
  File "/binbin/towhee-test-env-3.8-0.8.1/lib/python3.8/site-packages/pymilvus/decorators.py", line 85, in handler
    raise e
  File "/binbin/towhee-test-env-3.8-0.8.1/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler
    return func(self, *args, **kwargs)
  File "/binbin/towhee-test-env-3.8-0.8.1/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 483, in search
    return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, auto_id=auto_id, **kwargs)
  File "/binbin/towhee-test-env-3.8-0.8.1/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 452, in _execute_search_requests
    raise pre_err
  File "/binbin/towhee-test-env-3.8-0.8.1/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 443, in _execute_search_requests
    raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=1, reason=stream operation failed: [UnexpectedError] Assert "(index < size_)" at /go/src/github.com/milvus-io/milvus/internal/core/src/segcore/ConcurrentVector.h:51
```

### Expected Behavior

Range search runs successfully

### Steps To Reproduce

```python
from pymilvus import CollectionSchema, FieldSchema
from pymilvus import Collection
from pymilvus import connections
from pymilvus import DataType
from pymilvus import Partition
from pymilvus import utility

connections.connect()
dim = 128
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
float_field = FieldSchema(name="float", dtype=DataType.FLOAT)
bool_field = FieldSchema(name="bool", dtype=DataType.BOOL)
string_field = FieldSchema(name="string", dtype=DataType.VARCHAR, max_length=65535)
float_vector = FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=dim)
schema = CollectionSchema(fields=[int64_field, float_field, bool_field, float_vector])

collection = Collection("test_search_collection_binbin_tmp_0", schema=schema)
nb=3000
import random
import numpy as np
vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
res = collection.insert([[i for i in range(nb)], [np.float32(i) for i in range(nb)], [np.bool_(i) for i in range(nb)], vectors])
index_param = {"index_type": "IVF_FLAT", "metric_type": "L2", "params": {"nlist": 100}}
collection.create_index("float_vector", index_param, index_name="index_name_1")

collection.load()
default_search_params = {"metric_type": "L2", "params": {"nprobe": 10, "radius": 10, "range_filter": 5}}
nq = 2
limit = 10
res1 = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0")
```

**Comment by binbinlv** [comment]
Set it as urgent because it blocks "range search" function.

**Comment by binbinlv** [comment]
milvus log:
[22348.log](https://github.com/milvus-io/milvus/files/10802744/22348.log)

**Comment by smellthemoon** [comment]
<img width="1235" alt="1" src="https://user-images.githubusercontent.com/64083300/220803386-abfa4f4a-42fc-453e-8252-6e8c3f571ead.png">
<img width="838" alt="3" src="https://user-images.githubusercontent.com/64083300/220803564-824a08dd-1a97-45d9-892e-b858691d5806.png">
assert when get searchResults, when return to the table, offset returned by the range search exceeds the number of rows in the segment.
<img width="668" alt="4" src="https://user-images.githubusercontent.com/64083300/220803985-8b0f4012-7295-4a80-a62d-17938f15d7a1.png">
Do the same thing to deal with this situation as search.
<img width="797" alt="5" src="https://user-images.githubusercontent.com/64083300/220804395-ecbdd33c-0601-439c-9342-b32d52edc3c3.png">

**Comment by smellthemoon** [comment]
fixed with #22352

**Comment by binbinlv** [comment]
Fixed in :
milvus: bddda5bd (master latest)
pymilvus: 2.3.0.dev43
